apps/backend/src/services/compose_deployment.py

```
@@ -253,121 +251,115 @@ async def scan_docker_networks(self, request: NetworkScanRequest) -> NetworkScan
-                raise DeviceNotFoundError(hostname, "hostname")
-                
+            device = await svc_get_device_by_hostname(session, hostname)
             return device
 
-    def _parse_compose_content(self, content: str) -> Dict[str, Any]:
+    def _parse_compose_content(self, content: str) -> dict[str, Any]:
         """Parse docker-compose YAML content."""
         try:
             return yaml.safe_load(content)
         except yaml.YAMLError as e:
             raise ValidationError(f"Invalid docker-compose YAML: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)